### ISSUE TITLE
Erroneous error message when using F# 3.1 matrix slicing

### ISSUE THREAD

**Issue by Tarmil** [fsharp/fsharp-compiler-docs#116]
When passing the following code to the code service:

``` fsharp
let arr = Array2D.create 4 4 0
let a = arr.[1, *]
```

the following is reported:

```
The value, constructor, namespace or type 'GetArraySlice2DFixed1' is not defined
```

at the location of the dot in `arr.[1, *]`.

There is no error when using older kinds of slicing (such as `arr.[1.., *]`) nor other new F# 3.1 features (such as `type Foo = Foo of x:int`).

I get this result in .NET/Windows, but not on mono/ArchLinux. I checked with `procexp` on Windows that the correct versions (4.3.1.x) of `FSharp.Core.dll` and `FSharp.Core.Compiler.dll` are loaded.
